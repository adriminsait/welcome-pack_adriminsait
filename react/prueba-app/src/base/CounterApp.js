import { useState } from 'react';

import styles from './CounterApp.module.scss'

function CounterApp(value) {

    const[counter, setCounter] = useState(value.value);

    const handleAdd = () => {
        setCounter(counter + 1);
    }

    const handleSubtract = () => setCounter(counter - 1);

    const handleReset = () => setCounter(value.value)

    return (
        <div className={styles.main}>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>

            <div className={styles.buttonpad}>
                <button onClick={ handleAdd }>+</button>
                <button onClick={ handleReset }>Reset</button>
                <button onClick={ handleSubtract }>-</button>
            </div>
        </div>
    );
}

export default CounterApp;
  