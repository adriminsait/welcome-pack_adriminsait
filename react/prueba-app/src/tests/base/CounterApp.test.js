import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import CounterApp from '../../base/CounterApp';

describe('Pruebas en <CounterApp>', () => {

    const wrapper = shallow(<CounterApp value={10}/>);
    
    test('mostrar <CounterApp>', () => {
        
        const counterText = wrapper.find('h2').text().trim();

        expect(wrapper).toMatchSnapshot();
        expect(counterText).toBe('10');
    });

    test('boton +', () => {
        
        wrapper.find('button').at(0).simulate('click', {});
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('11');

    });

    test('boton -', () => {
        
        wrapper.find('button').at(2).simulate('click', {});
        wrapper.find('button').at(2).simulate('click', {});
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('9');

    });
    
    test('reset', () => {
        
        wrapper.find('button').at(2).simulate('click', {});
        wrapper.find('button').at(1).simulate('click', {});

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('10');

    });

});
