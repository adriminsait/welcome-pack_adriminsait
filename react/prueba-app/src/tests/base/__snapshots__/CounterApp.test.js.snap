// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp> mostrar <CounterApp> 1`] = `
<div
  className="main"
>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <div
    className="buttonpad"
  >
    <button
      onClick={[Function]}
    >
      +
    </button>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      onClick={[Function]}
    >
      -
    </button>
  </div>
</div>
`;
